service/history/ndc_task_util.go

```diff
@@ -177,6 +192,32 @@ func LoadMutableStateForTask(
 	return mutableState, nil
 }
 
+func validateTaskByClock(
+	shardContext shard.Context,
+	task tasks.Task,
+) error {
+	shardID := shardContext.GetShardID()
+	taskClock := vclock.NewVectorClock(
+		shardContext.GetClusterMetadata().GetClusterID(),
+		shardContext.GetShardID(),
+		task.GetTaskID(),
+	)
+	currentClock := shardContext.CurrentVectorClock()
+	result, err := vclock.Compare(taskClock, currentClock)
+	if err != nil {
+		return err
+	}
+	if result >= 0 {
+		shardContext.UnloadForOwnershipLost()
+		return &persistence.ShardOwnershipLostError{
+			ShardID: shardID,
+			Msg:     fmt.Sprintf("Shard: %v task clock validation failed, reloading", shardID),
```

Comment:
So we've lost ownership if taskClock is >= currentClock?

Reply:
yes. Because that means the task is generated by another newer owner of this shard